src/server/cmd/pachd/main.go

```diff
@@ -711,25 +712,25 @@ func doFullMode(ctx context.Context, config interface{}) (retErr error) {
 	}))
 	eg.Go(maybeIgnoreErrorFunc("S3 Server", requireNoncriticalServers, func() error { return s3Server{env.GetPachClient, env.Config().S3GatewayPort}.listenAndServe(ctx) }))
 	eg.Go(maybeIgnoreErrorFunc("Prometheus Server", requireNoncriticalServers, func() error { return prometheusServer{port: env.Config().PrometheusPort}.listenAndServe(ctx) }))
-	go func(c chan os.Signal) {
-		<-c
-		log.Println("terminating; waiting for pachd server to gracefully stop")
-		var g, _ = errgroup.WithContext(ctx)
-		g.Go(func() error { externalServer.Server.GracefulStop(); return nil })
-		g.Go(func() error { internalServer.Server.GracefulStop(); return nil })
-		if err := g.Wait(); err != nil {
-			log.Errorf("error waiting for pachd server to gracefully stop: %v", err)
+	eg.Go(func() error {
+		<-ctx.Done() // wait for main context to complete
+		var eg = new(errgroup.Group)
```

Comment:
what do you think of having a timeout for Gracefully stopping?

Reply:
Well, Kubernetes should stop it anyway, but … it’s not a bad idea. Do you think it should be configurable or just a hardcoded 10, 15 or 60 seconds?

Reply:
I think hardcoding it is fine

Reply:
I think 10 seconds is reasonable